bundles/core/org.eclipse.smarthome.core.ephemeris/src/main/java/org/eclipse/smarthome/core/ephemeris/EphemerisManager.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.ephemeris;
+
+import java.net.MalformedURLException;
+
+/**
+ * This service provides functionality around days of the year and is the central
+ * service to be used directly by others.
+ *
+ * @author Gaël L'hopital - Initial contribution and API
+ *
+ */
+public interface EphemerisManager {
+
+    boolean isWeekEnd();
+
+    boolean isWeekEnd(int offset);
+
+    boolean isBankHoliday();
+
+    boolean isBankHoliday(int offset);
+
+    String getBankHolidayName();
+
+    String getBankHolidayName(int offset);
+
+    String getHolidayUserFile(String filename) throws MalformedURLException;
+
+    String getHolidayUserFile(int offset, String filename) throws MalformedURLException;
```

Comment:
It would be helpful if the methods on this interface also have JavaDocs.
Without these docs you'll have to look at the implementation to see how the offsets are used.